### PR TITLE
Add builtin function snippets

### PR DESCRIPTION
**Pull request by weeman1337**
To be honest I don't really have a clue what I am doing here ;)  
I found that the "NameSearcher" doesn't add snippets → is now implemented.

Relates to https://github.com/phpactor/phpactor/issues/1230

Once this has been released I will file a PR with an update for "completion-worse-extension".

### CODE REVIEW

**Comment by weeman1337** [comment]
:( The failed tests have been there from the beginning of my work. Is the current `master` broken?

File: lib/Core/Completor/NameSearcherCompletor.php [hunk 1]
```diff
@@ -52,13 +71,22 @@ protected function createSuggestion(NameSearchResult $result, array $options = [
...
+            'short_description' => $name,
             'type' => $this->suggestionType($result),
             'class_import' => $this->classImport($result),
-            'name_import' => $result->name()->__toString(),
-            'priority' => $this->prioritizer->priority($result->uri(), $sourceUri)
+            'name_import' => $name,
+            'priority' => $this->prioritizer->priority($result->uri(), $sourceUri),
         ];
+
+        if ($result->type()->isFunction()) {
```
Comment (dantleech) [inline lib/Core/Completor/NameSearcherCompletor.php:84]: this is a bit of a smell, could we not put this logic in the FunctionCompletor?

**Comment by dantleech** [comment]
cool, thanks. the tests are failing because the package depends on `phpbench@dev-master` which recently got a bunch of breaking changes, you could change it to `^1.0.0@alpha` for the benefit of the PR.

left a comment, not sure the function-logic belongs in the abstract class, would it be possible to move it to the function completor? or is this the best place?

cc @camilledejoye

File: lib/Core/Completor/NameSearcherCompletor.php (hunk 1 above)
Comment (weeman1337) [reply lib/Core/Completor/NameSearcherCompletor.php:84]: [This](https://github.com/phpactor/completion/blob/master/lib/Bridge/TolerantParser/WorseReflection/WorseFunctionCompletor.php) FunctionCompletor? Somehow for example `in_array` is found by the `NameSearcherCompletor`.

Comment (dantleech) [reply lib/Core/Completor/NameSearcherCompletor.php:84]: ah that makes sense.

Comment (dantleech) [reply lib/Core/Completor/NameSearcherCompletor.php:84]: we could also add a dedicated `NameSearchResultSnippetFormatter`. But for now I think your change is fine, espeicially as we already have `type()->isClass` in the `classImport` method.

**Comment by weeman1337** [comment]
> cool, thanks. the tests are failing because the package depends on phpbench@dev-master which recently got a bunch of breaking changes, you could change it to ^1.0.0@alpha for the benefit of the PR.

Changing that doesn't help with the failing test :(

File: lib/Core/Completor/NameSearcherCompletor.php (hunk 1 above)
Comment (weeman1337) [reply lib/Core/Completor/NameSearcherCompletor.php:84]: Unfortunately the `Formatter` interface requires `format` to return a `string`. I cannot return `null` there. Or I have to create a separate standalone class for that.

**Comment by weeman1337** [comment]
See https://github.com/phpactor/completion/pull/51

File: lib/Core/Completor/NameSearcherCompletor.php (hunk 1 above)
Comment (dantleech) [reply lib/Core/Completor/NameSearcherCompletor.php:84]: The `SnippetFormatter` has a `canFormat($object)` method so would look like `if (!$object instanceof NameSearchResult) { return false; } return $object->isFunction()`). Like I say it's up to you, just an idea.

**Comment by dantleech** [comment]
Thanks, I merged the other PR and fixed the `phpbench.json` config, so you can `git fetch` `git rebase origin/master` and it should be green ...

File: lib/Core/Completor/NameSearcherCompletor.php (hunk 1 above)
Comment (weeman1337) [reply lib/Core/Completor/NameSearcherCompletor.php]: I have implemented the `NameSearchResultFunctionSnippetFormatter` as suggested. In addition to that I moved the code to the `NameSearcherCompletor`. In this way the `DoctrineAnnotationCompletor` is not affected by the changes.

**Comment by dantleech** [comment]
Thanks :)

**Comment by weeman1337** [comment]
Wow - that was fast :rocket: 

I am now going to file the PRs for the extention repos.

**Comment by weeman1337** [comment]
Okay - looks like I have to wait for the new Tag.

**Comment by dantleech** [comment]
You can depend on the next tag, as this is a non-breaking change require `0.4.5` - composer is configured to prefer-stable but will pull in dev-master otherwise (`0.4.x`)

**Pull request merged**